packages/thirdweb/src/react/web/ui/TransactionButton/index.tsx

```diff
@@ -136,6 +136,8 @@ export function TransactionButton(props: TransactionButtonProps) {
 
           if (onTransactionConfirmed) {
             const receipt = await doWaitForReceipt(result);
+            if (receipt.status === "reverted")
+              throw new Error("Execution reverted");
```

Comment:
interesting - it didn't throw on simulation?

what was the actual error? out of gas?

Reply:
would be nice to add the tx hash in here